pkg/analysis_server/lib/src/services/correction/fix_internal.dart

```diff
@@ -1875,6 +1884,25 @@ class FixProcessor {
     }
   }
 
+  void _addFix_removeTypeName() {
+    final FunctionTypedFormalParameter functionTyped =
+        node is FunctionTypedFormalParameter
+            ? node
+            : node.getAncestor((node) => node is FunctionTypedFormalParameter);
+    if (functionTyped != null) {
+      _addReplaceEdit(rf.rangeNode(functionTyped),
+          utils.getNodeText(functionTyped.identifier));
+      _addFix(DartFixKind.REMOVE_TYPE, []);
+      return;
+    }
```

Comment:
Yes, you are right, sorry for making you waste your time, I mixed the rules in my mind, so yes, AvoidAnnotatingWithDynamic can crash with this fix, avoid_types_on_closure_parameters is the one that do that what I said.

I did not create the tests because I did not know how to do it without a lint, but know how you helped me to figure out what happened can create it will AvoidAnnotatingWithDynamic and fix the fix, thanks!